Framework/Algorithms/src/Q1DWeighted.cpp

```diff
@@ -162,6 +207,167 @@ void Q1DWeighted::bootstrap(const MatrixWorkspace_const_sptr &inputWS) {
   m_normalisation = m_intensities;
 }
 
+/**
+ * @brief Q1DWeighted::getTableShapes
+ * if the user provided a shape table, parse the stored values and get the
+ * viewport and the sector shapes
+ */
+void Q1DWeighted::getTableShapes() {
+  ITableWorkspace_sptr shapeWs = getProperty("ShapeTable");
+  size_t rowCount = shapeWs->rowCount();
+
+  std::map<std::string, std::vector<double>> viewportParams;
+
+  // by convention, the last row is supposed to be the viewport
+  getViewportParams(shapeWs->String(rowCount - 1, 1), viewportParams);
+
+  for (size_t i = 0; i < rowCount - 1; ++i) {
+    std::map<std::string, std::vector<std::string>> paramMap;
+    std::vector<std::string> splitParams;
+    boost::algorithm::split(splitParams, shapeWs->String(i, 1),
+                            boost::algorithm::is_any_of("\n"));
+    std::vector<std::string> params;
+
+    for (std::string val : splitParams) {
+      if (val.empty())
+        continue;
+      boost::algorithm::split(params, val, boost::algorithm::is_any_of("\t"));
+
+      // NB : the first value of the vector also is the key, and is not a
+      // meaningful value
+      paramMap[params[0]] = params;
+    }
+    if (paramMap["Type"][1] == "sector")
+      getSectorParams(paramMap["Parameters"], viewportParams);
+    else
+      g_log.information("Non-supported shape found in the table. Ignored.");
+  }
+}
+
+/**
+ * @brief Q1DWeighted::getViewportParams
+ * get the parameters defining the viewport of the instrument view when the
+ * shapes were created, and store them in a map
+ * @param viewport the parameters as they were saved in the shape table
+ * @param viewportParams the map to fill
+ */
+void Q1DWeighted::getViewportParams(
+    std::string &viewport,
+    std::map<std::string, std::vector<double>> &viewportParams) {
+  std::vector<std::string> params;
+  boost::algorithm::split(params, viewport,
+                          boost::algorithm::is_any_of("\t, \n"));
+
+  if (params[0] != "Translation") {
+    g_log.error("No viewport found in the shape table. Please provide a table "
+                "using shapes drawn in the Full3D projection.");
+  }
+
+  // Translation
+  viewportParams[params[0]] = std::vector<double>(2);
+  viewportParams[params[0]][0] = std::stod(params[1]);
+  viewportParams[params[0]][1] = std::stod(params[2]);
+
+  // Zoom
+  viewportParams[params[3]] = std::vector<double>(1);
+  viewportParams[params[3]][0] = std::stod(params[4]);
+
+  // Rotation quaternion
+  viewportParams[params[5]] = std::vector<double>(4);
+  viewportParams[params[5]][0] = std::stod(params[6]);
+  viewportParams[params[5]][1] = std::stod(params[7]);
+  viewportParams[params[5]][2] = std::stod(params[8]);
+  viewportParams[params[5]][3] = std::stod(params[9]);
+
+  if (fabs(viewportParams["Rotation"][0]) > 1e-10 ||
```

Comment:
Define the tolerance once in a constant. 
Also should be `std::fabs` to be consistent with the rest.